zipline/src/commonMain/kotlin/app/cash/zipline/ZiplineScoped.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2022 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline
+
+/**
+ * Implement this interface on inbound services to specify which scope service arguments are added
+ * to. See [ZiplineScope] for details.
+ *
+ * Do not extend this in your [ZiplineService] interfaces; it should only be implemented by concrete
+ * implementation classes.
```

Comment:
Can the compiler validate this? At least on dealing with calls to `take` walk its interface hierarchy and validate none do this.

Reply:
Absolutely. On the list.

Reply:
https://github.com/cashapp/zipline/issues/775